File: web/src/api/pluginDebug.ts
import client, { unwrap } from './client'

export interface PluginDebugLogItem {
  name: string
  runId: string
  kind: string
  ok: boolean
  error?: string
  durationMs: number
  version: string
  shopId: number
  shopName: string
  receivedAt: string
  size: number
}

export interface PluginDebugEvent {
  ms: number
  at: string
  level: string
  step: string
  data?: unknown
}

export interface PluginDebugLogRecord {
  runId: string
  kind: string
  ok: boolean
  error?: string
  durationMs: number
  version: string
  shopId: number
  shopName: string
  receivedAt: string
  meta?: Record<string, unknown>
  events: PluginDebugEvent[]
}

export async function fetchPluginDebugLogs() {
  return unwrap<{ dir: string; list: PluginDebugLogItem[] }>(await client.get('/plugin-debug-logs'))
}

export async function fetchPluginDebugLog(name: string) {
  return unwrap<PluginDebugLogRecord>(
    await client.get(`/plugin-debug-logs/${encodeURIComponent(name)}`),
  )
}

File: internal/dto/plugin_debug.go
package dto

type PluginDebugEvent struct {
	Ms    int64       `json:"ms"`
	At    string      `json:"at"`
	Level string      `json:"level"`
	Step  string      `json:"step"`
	Data  interface{} `json:"data,omitempty"`
}

type PluginDebugLogInput struct {
	RunID      string                 `json:"runId"`
	Kind       string                 `json:"kind"`
	OK         bool                   `json:"ok"`
	Error      string                 `json:"error"`
	DurationMs int64                  `json:"durationMs"`
	Version    string                 `json:"version"`
	Meta       map[string]interface{} `json:"meta"`
	Events     []PluginDebugEvent     `json:"events"`
}

type PluginDebugLogResult struct {
	Name string `json:"name"`
	Dir  string `json:"dir"`
}
